mne/decoding/time_frequency.py

```diff
@@ -0,0 +1,161 @@
+    ``mne.time_frequency.tfr_multitaper``
+    """
+
+    def __init__(self, frequencies, sfreq=1.0, method='morlet', n_cycles=7.0,
+                 zero_mean=None, time_bandwidth=None, use_fft=True, decim=1,
+                 output='complex', n_jobs=1, verbose=None):
+        """Init TimeFrequency transformer."""
+        frequencies, sfreq, zero_mean, n_cycles, time_bandwidth, decim = \
+            _check_tfr_param(frequencies, sfreq, method, zero_mean, n_cycles,
+                             time_bandwidth, use_fft, decim, output)
+        self.frequencies = frequencies
+        self.sfreq = sfreq
+        self.method = method
+        self.n_cycles = n_cycles
+        self.zero_mean = zero_mean
+        self.time_bandwidth = time_bandwidth
+        self.use_fft = use_fft
+        self.decim = decim
+        # Check that output is not an average metric (e.g. ITC)
+        if output not in ['complex', 'power', 'phase']:
+            raise ValueError("output must be 'complex', 'power', 'phase'.")
+        self.output = output
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def fit_transform(self, X, y=None):
+        """
+        Time-frequency transform of times series along the last axis.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_channels, n_times)
+            The training data samples. The channel dimension can be zero- or
+            n-dimensional.
+        y : None
+            For scikit-learn compatibility purposes.
+
+        Returns
+        -------
+        Xt : array, shape (n_samples, n_channels, n_frequencies, n_times)
+            The time-frequency transform of the data, where n_channels can be
+            zero- or n-dimensional.
+        """
+        return self.fit(X, y).transform(X)
+
+    def fit(self, X, y=None):
+        """ Does nothing, for scikit-learn compatibility purposes.
+
+        Parameters
+        ----------
+        X : None
+            The target data.
+        y : None
+            The target values.
+
+        Returns
+        -------
+        self : object
+            Return self.
+        """
+        return self
+
+    def transform(self, X):
+        """Time-frequency transform of times series along the last axis.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_channels, n_times)
+            The training data samples. The channel dimension can be zero- or
+            n-dimensional.
+
+        Returns
+        -------
+        Xt : array, shape (n_samples, n_channels, n_frequencies, n_times)
+            The time-frequency transform of the data, where n_channels can be
+            zero- or n-dimensional.
+
+        """
+        # Ensure 3-dimensional X
+        n_samples, n_times = len(X), X.shape[-1]
+        shape = X.shape[1:-1]
+        if not shape:
+            X = X[:, np.newaxis, :]
+        elif len(shape) > 1:
+            X = X.reshape(n_samples, -1, n_times)
```

Comment:
if you have an ECoG grid, you may want to keep the array dimension (n_chans * n_chans) for easy 2D interpolation in subsequent steps.

Too specific? I'm happy to remove it